projects/packages/videopress/src/client/admin/components/video-quick-actions/index.tsx

```diff
@@ -237,35 +238,53 @@ export const ConnectVideoQuickActions = ( props: ConnectVideoQuickActionsProps )
 	const { data, updateVideoPrivacy, deleteVideo, isUpdatingPrivacy } = useVideo( videoId );
 
 	const [ showDeleteModal, setShowDeleteModal ] = useState( false );
-	const [ updatingThumbnail, setUpdatingThumbnail ] = useState( false );
+	const [ updatingPosterImage, setUpdatingPosterImage ] = useState( false );
 	const {
 		frameSelectorIsOpen,
 		handleCloseSelectFrame,
 		handleOpenSelectFrame,
 		handleVideoFrameSelected,
 		selectedTime,
 		handleConfirmFrame,
-		updatePosterImage,
+		updatePosterImage: updatePosterImageFromFrame,
 	} = usePosterEdit( { video: data } );
+	const { updatePosterImage: updatePosterImageFromLibrary } = useLibraryPosterImage( {
+		video: data,
+	} );
+
+	const updatePosterImage = async updateRequest => {
+		try {
+			setUpdatingPosterImage( true );
+			const posterImage = await updateRequest();
+
+			if ( posterImage == null ) {
+				return;
+			}
+
+			const videoData = { ...data, posterImage };
+			dispatch?.setVideo( videoData );
```

Comment:
I think we should create a new action for this considering that we'd like to show an uploading/updating state